libafl/src/executors/timeout.rs

```diff
@@ -104,17 +109,32 @@ impl<E> TimeoutExecutor<E> {
             itimerval,
         }
     }
+}
 
-    #[cfg(windows)]
+#[cfg(windows)]
+impl<E: HasInProcessHandlers> TimeoutExecutor<E> {
     pub fn new(executor: E, exec_tmout: Duration) -> Self {
-        let milli_sec = exec_tmout.as_millis() as u32;
-        let timer_queue = unsafe { CreateTimerQueue() };
-        let ph_new_timer = HANDLE::NULL;
+        let milli_sec = exec_tmout.as_millis() as i64;
+        let timeout_handler: PTP_TIMER_CALLBACK =
+            unsafe { std::mem::transmute(executor.inprocess_handlers().timeout_handler) };
+        let tp_timer = unsafe {
+            CreateThreadpoolTimer(
+                Some(timeout_handler),
+                &mut GLOBAL_STATE as *mut _ as *mut c_void,
+                &TP_CALLBACK_ENVIRON_V3::default(),
+            )
+        };
+        let mut critical = RTL_CRITICAL_SECTION::default();
```

Comment:
the critical section should be created before the timer, right? Else there's a race condition here?

Reply:
CreateThreadpoolTimer and InitializeCriticalSection are both the initializations of the instances.
we _enable_ them in run_target with SetThreadpoolTimer and EnterCriticalSection, respectively.